packages/core/src/browser/frontend-application.ts

```diff
@@ -138,13 +147,15 @@ export class FrontendApplication {
         await animationFrame();
         this.stateService.state = 'attached_shell';
 
-        await this.initializeLayout();
+        await this.measure('initializeLayout', () => this.initializeLayout(), 'Initialize the workbench layout', false);
         this.stateService.state = 'initialized_layout';
         await this.fireOnDidInitializeLayout();
 
         await this.revealShell(host);
         this.registerEventListeners();
         this.stateService.state = 'ready';
+
+        startup?.then(idToken => this.backendStopwatch!.stop(idToken, 'Frontend application start', []));
```

Comment:
This needs to be an optional execution as the `backendStopwatch` is also marked as optional. Otherwise this will crash when the `backendStopwatch` is not injected
```suggestion
        startup?.then(idToken => this.backendStopwatch?.stop(idToken, 'Frontend application start', []));
```

Reply:
But we wouldn't have a `startup` if the `backendStopwatch` wasn't injected, so we can be sure that the latter exists. That's why I used the `!`.

Reply:
Hmm yes you're right, I overlooked that. Usually I like to add a comment whenever we're bypassing TS, so I would add something like:
```
// backendStopwatch must exist when startup exists
```

Reply:
Yup. If it raises its head in review, it probably at least needs a comment! 😉